intellij/src/saros/intellij/editor/LocalEditorHandler.java

```diff
@@ -169,29 +181,44 @@ public void removeEditor(@NotNull SPath path) {
   /**
    * Saves the document under path, thereby flushing its contents to disk.
    *
+   * <p>Does nothing if there is no unsaved document content for the given resource. This is the
+   * case if the resource document has not been modified or the resource can not be opened as a
+   * document.
+   *
    * @param path the path for the document to save
    * @see Document
    */
   public void saveDocument(@NotNull SPath path) {
 
     Document document = editorPool.getDocument(path);
 
-    if (document == null) {
-      VirtualFile file = VirtualFileConverter.convertToVirtualFile(path);
-
-      if (file == null || !file.exists()) {
-        LOG.warn("Failed to save document for " + path + " - could not get a valid VirtualFile");
+    if (document != null) {
```

Comment:
Wouldn't it be better to check if the document is null here and then start to retrieve it ? Currently the save logic is present twice.

Reply:
That is how it was done with the old implementation. I thought that this way was easier to read this way as there are two distinct logical paths through the code. The first part from l.195 to l.202 deals with the case that the document is known to the editor pool and the rest deals with the case that the document has to be requested.

But I can change it again if you prefer. That would then lead to the "has unsaved changes" check being run twice if the document can not be obtained from the editor pool.

This is due to the check in l.212 of whether the file has unsaved changes being used to filter out resources that can't be represented by a document and for which the `getDocument(...)` call in l.216 would therefore always return `null`.

Another approach would be to accept such cases and reduce the log level to debug in l.219.